### PR TITLE
fix: validate plugins : allow many versions of the same plugin

### PR DESCRIPTION
**Pull request by camilamacedo86**
closes: #1534

### CODE REVIEW

File: pkg/cli/cli.go [hunk 1]
```diff
@@ -351,12 +351,13 @@ func validatePlugins(plugins ...plugin.Base) error {
 					pluginName, projectVersion, err)
 			}
 		}
-		// Check for duplicate plugin names. Names outside of a version can
+		// Check if has duplicate plugins. Names outside of a version can
```
Comment (estroz) [inline pkg/cli/cli.go:354]:
```suggestion
		// Check for duplicate plugin keys.
```

File: pkg/cli/cli.go [hunk 2]
```diff
@@ -351,12 +351,13 @@ func validatePlugins(plugins ...plugin.Base) error {
...
 		// conflict because multiple project versions of a plugin may exist.
```
Comment (estroz) [inline pkg/cli/cli.go:355]:
```suggestion
```

File: pkg/cli/cli.go [hunk 3]
```diff
@@ -351,12 +351,13 @@ func validatePlugins(plugins ...plugin.Base) error {
...
-		if _, seen := pluginNameSet[pluginName]; seen {
-			return fmt.Errorf("two plugins have the same name: %q", pluginName)
+		pluginNameVersion := fmt.Sprintf("%v/%v", p.Name(), p.Version())
```
Comment (estroz) [inline pkg/cli/cli.go:356]:
```suggestion
		pluginKey := plugin.KeyFor(p)
```

File: pkg/cli/cli.go [hunk 4]
```diff
@@ -351,12 +351,13 @@ func validatePlugins(plugins ...plugin.Base) error {
...
+		if _, seen := pluginNameSet[pluginNameVersion]; seen {
```
Comment (estroz) [inline pkg/cli/cli.go:357]:
```suggestion
		if _, seen := pluginNameSet[pluginKey]; seen {
```

File: pkg/cli/cli.go [hunk 5]
```diff
@@ -351,12 +351,13 @@ func validatePlugins(plugins ...plugin.Base) error {
...
+			return fmt.Errorf("two plugins have the same name and version: %q", pluginNameVersion)
 		}
-		pluginNameSet[pluginName] = struct{}{}
+		pluginNameSet[pluginNameVersion] = struct{}{}
```
Comment (estroz) [inline pkg/cli/cli.go:360]:
```suggestion
		pluginNameSet[pluginKey] = struct{}{}
```

File: pkg/cli/cli.go [hunk 6]
```diff
@@ -351,12 +351,13 @@ func validatePlugins(plugins ...plugin.Base) error {
...
 			}
 		}
-		// Check for duplicate plugin names. Names outside of a version can
+		// Check if has duplicate plugins. Names outside of a version can
 		// conflict because multiple project versions of a plugin may exist.
-		if _, seen := pluginNameSet[pluginName]; seen {
-			return fmt.Errorf("two plugins have the same name: %q", pluginName)
+		pluginNameVersion := fmt.Sprintf("%v/%v", p.Name(), p.Version())
+		if _, seen := pluginNameSet[pluginNameVersion]; seen {
+			return fmt.Errorf("two plugins have the same name and version: %q", pluginNameVersion)
```
Comment (estroz) [inline pkg/cli/cli.go:358]:
```suggestion
			return fmt.Errorf("two plugins have the same key: %q", pluginKey)
```

File: pkg/cli/cli_test.go [hunk 7]
```diff
@@ -78,6 +85,28 @@ var _ = Describe("CLI", func() {
...
+
+	Describe("Check if has duplicate plugins", func() {
+		It("should work successfully when a plugin has many versions", func() {
+			plugins := makePluginsForKeys(
+				"foo.example.com/v1.0.0",
+				"foo.example.com/v2.0.0",
+				"foo.example.com/v3.0.0",
+			)
+
+			err := validatePlugins(plugins[0], plugins[1], plugins[2])
```
Comment (estroz) [inline pkg/cli/cli_test.go:97]:
```suggestion
			err := validatePlugins(plugins...)
```

File: pkg/cli/cli_test.go [hunk 8]
```diff
@@ -78,6 +85,28 @@ var _ = Describe("CLI", func() {
...
+			Expect(err).NotTo(HaveOccurred())
+		})
+		It("should fail when found more than one plugin with the same name and version", func() {
+			plugins := makePluginsForKeys(
+				"foo.example.com/v1.0.0",
+				"foo.example.com/v1.0.0",
+			)
+
+			err := validatePlugins(plugins[0], plugins[1])
```
Comment (estroz) [inline pkg/cli/cli_test.go:106]:
```suggestion
			err := validatePlugins(plugins...)
```

File: pkg/cli/cli_test.go (hunk 7 above)
Comment (camilamacedo86) [reply pkg/cli/cli_test.go]: Good catcher 👍

File: pkg/cli/cli_test.go (hunk 8 above)
Comment (camilamacedo86) [reply pkg/cli/cli_test.go]: Good catcher 👍

**Pull request merged**